app/src/main/java/de/hhn/aib/labsw/blackmirror/view/widgets/AbstractWidget.java

```diff
@@ -30,6 +30,40 @@ public void setPosition(Position pos) {
                     screen.height / 2 - this.getHeight() / 2);
         }
     }
-}
 
+    public void setPosition(int xPosition, int yPosition) {
+        Dimension screen = Toolkit.getDefaultToolkit().getScreenSize();
+        Dimension pos = new Dimension(xPosition, yPosition);
+
+        if(pos.height == 1) {
+            if(pos.width == 1) {
+                this.setLocation(0, 0);
+            } else if(pos.width == 2) {
+                this.setLocation(screen.width/2 - this.getWidth(), 0);
+            } else {
+                this.setLocation(screen.width - this.getWidth(), 0);
+            }
+        }
+
+        if(pos.height == 2) {
+            if(pos.width == 1) {
+                this.setLocation(0, screen.height/2 - this.getHeight()/2);
+            } else if(pos.width == 2) {
+                this.setLocation(screen.width/2 - this.getWidth(), screen.height/2 - this.getHeight()/2);
+            } else {
+                this.setLocation(screen.width - this.getWidth(), screen.height/2 - this.getHeight()/2);
+            }
+        }
+
+        if(pos.height == 3) {
+            if(pos.width == 1) {
+                this.setLocation(0, screen.height - this.getHeight());
+            } else if(pos.width == 2) {
+                this.setLocation(screen.width/2 - this.getWidth(), screen.height - this.getHeight());
+            } else {
+                this.setLocation(screen.width - this.getWidth(), screen.height - this.getHeight());
+            }
+        }
```

Comment:
There is no need to repeat yourself all the time here. Just calculate the x and y position seperately and assign them (so that you have to seperate if-else-blocks with three branches each). As soon as we have consistenly sized widgets we should precalculate all the "slots" and simply assign them.